OpenRA.Mods.Common/Traits/Conditions/GrantConditionWhileAiming.cs

```diff
@@ -0,0 +1,55 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class GrantConditionWhileAimingInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[GrantedConditionReference]
+		[Desc("The condition to grant while aiming.")]
+		public readonly string Condition = null;
+
+		object ITraitInfo.Create(ActorInitializer init) { return new GrantConditionWhileAiming(this); }
+	}
+
+	public class GrantConditionWhileAiming : INotifyCreated, INotifyAiming
```

Comment:
Thinking out loud, maybe a job for a different PR: if `INotifyAiming` provided some information about *what* was being targeted then we could add a filter (probably using target types), which would solve the plumbing for #2125.